src/cli/mod.rs

```diff
@@ -0,0 +1,53 @@
+mod serve;
+
+use clap::{Args, Parser, Subcommand};
+
+use crate::domain::{
+    config::Config,
+    error::{LoadBalancerError, LoadBalancerResult},
+    worker::WorkersList,
+};
+
+#[derive(Parser)]
+#[command(version, about, long_about = None)]
+struct Cli {
+    // #[arg(short, long, action = clap::ArgAction::Count)]
+    // debug: u8
+    #[command(subcommand)]
+    commands: Commands,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    /// Starts the load balancer
+    Serve(Serve),
+}
+
+#[derive(Args)]
+struct Serve {
+    /// Specify the port to run the server on.
+    #[clap(short, long, default_value = "1337")]
+    port: u16,
+
+    #[clap(short, long)]
+    worker: Vec<String>,
+}
+
+impl TryFrom<Serve> for Config {
```

Comment:
Look's like a convenient way to get Config! I haven't thought about this approach.